tests/unit/eblob_wrapper.h

```diff
@@ -0,0 +1,120 @@
+	: key(key_)
+	, value(std::move(value_))
+	{
+		memcpy(hashed_key.id, hashed_key_.id, EBLOB_ID_SIZE);
+	}
+
+	bool checked = false;
+	bool removed = false;
+	int key;
+	eblob_key hashed_key;
+	std::vector<char> value;
+
+	bool operator< (const Item &rhs) const {
+		return eblob_id_cmp(hashed_key.id, rhs.hashed_key.id) < 0;
+	}
+};
+
+
+class eblob_wrapper {
+public:
+	eblob_wrapper()
+	: data_dir_template_("/tmp/eblob-test-XXXXXX")
+	, data_dir_{mkdtemp(&data_dir_template_.front())}
+	, data_path_{data_dir_ + "/data"}
+	, log_path_{data_dir_ + "/log.log"}
+	, logger_{log_path_.c_str(), EBLOB_LOG_DEBUG} {
+		default_config.blob_flags = EBLOB_L2HASH | EBLOB_DISABLE_THREADS | EBLOB_AUTO_INDEXSORT;
+		default_config.sync = -2;
+		default_config.log = logger_.log();
+		default_config.file = (char *)data_path_.c_str();
+		default_config.blob_size = EBLOB_BLOB_DEFAULT_BLOB_SIZE;
+		default_config.records_in_blob = EBLOB_BLOB_DEFAULT_RECORDS_IN_BLOB;
+		default_config.defrag_percentage = EBLOB_DEFAULT_DEFRAG_PERCENTAGE;
+		default_config.defrag_timeout = EBLOB_DEFAULT_DEFRAG_TIMEOUT;
+		default_config.index_block_size = EBLOB_INDEX_DEFAULT_BLOCK_SIZE;
+		default_config.index_block_bloom_length = EBLOB_INDEX_DEFAULT_BLOCK_BLOOM_LENGTH;
+		default_config.blob_size_limit = UINT64_MAX;
+		default_config.defrag_time = EBLOB_DEFAULT_DEFRAG_TIME;
+		default_config.defrag_splay = EBLOB_DEFAULT_DEFRAG_SPLAY;
+		default_config.periodic_timeout = EBLOB_DEFAULT_PERIODIC_THREAD_TIMEOUT;
+		default_config.stat_id = 12345;
+		default_config.chunks_dir = (char*)data_dir_.c_str();
+	}
+
+	void start(eblob_config *config = nullptr) {
+		if (config != nullptr)
+			backend_ = eblob_init(config);
+		else
+			backend_ = eblob_init(&default_config);
+	}
+
+	void restart(eblob_config *config = nullptr) {
+		stop();
+		start(config);
+	}
+
+	void stop() {
+		if (backend_) {
+			eblob_cleanup(backend_);
+			backend_ = nullptr;
+		}
+	}
+
+	~eblob_wrapper() {
+		stop();
+		boost::filesystem::remove_all(data_dir_);
+	}
+
+	eblob_backend *get() { return backend_; }
+	const eblob_backend *get() const { return backend_; }
+
+	Item generate_item(int key, const std::vector<char> &data) {
+		struct eblob_key hashed_key;
+		eblob_hash(get(), hashed_key.id, sizeof(hashed_key.id), &key, sizeof(key));
+		return Item(key, hashed_key, data);
+	}
+
+
+	int insert_item(Item &item) {
+		return eblob_write(get(), &item.hashed_key, item.value.data(), /* offset */ 0, item.value.size(), BLOB_DISK_CTL_APPEND);
+	}
+
+	int remove_item(Item &item) {
+		item.removed = true;
+		return eblob_remove_hashed(get(), &item.key, sizeof(item.key));
+	}
+
+public:
+	eblob_config default_config;
+
+private:
+	std::string data_dir_template_;
+	std::string data_dir_;
+	std::string data_path_;
+	std::string log_path_;
+	ioremap::eblob::eblob_logger logger_;
+	eblob_backend *backend_ = nullptr;
+};
+
+eblob_key hash(std::string key) {
+	eblob_key ret;
+	sha512_buffer(key.data(), key.size(), ret.id);
+	return ret;
+}
```

Comment:
Code style: blank line should be added before eof (I'm not sure, ask @shaitan)

Reply:
It was necessary some time ago and now after C++11 it's not.
https://stackoverflow.com/questions/72271/no-newline-at-end-of-file-compiler-warning

Reply:
No, I about code style. Everywhere we use newline at the end, because it's friendly to VCS.